WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignSectionLoader.swift

```diff
@@ -2,13 +2,34 @@ import Foundation
 import WordPressKit
 
 struct SiteDesignSectionLoader {
+    typealias Assembler = ((SiteIntentVertical?) -> [SiteDesignSection])
 
-    /// Fetches and assembles `SiteDesignSection`s from the API.
+    /// Builds a site design section assembler based on data fetched from the API.
     ///
-    /// - Parameters:
-    ///   - vertical: An optional Site Intent vertical.
-    ///   - completion: The result closure.
-    static func fetchSections(vertical: SiteIntentVertical?, completion: @escaping (Result<[SiteDesignSection], Error>) -> Void) {
+    /// - Parameter completion: A closure providing an assembler function or an error.
+    static func buildAssembler(completion: @escaping (Result<Assembler, Error>) -> Void) {
+        fetchRemoteDesigns { result in
+            switch result {
+            case .success(let remoteDesigns):
+                let categorySections = getCategorySectionsForRemoteSiteDesigns(remoteDesigns)
+
+                completion(.success({ vertical in
```

Comment:
I might totally be missing something here, but it seems that we are using a closure (`Assembler`) to return a result to `SiteDesignContentCollectionViewController`, in order to correctly populate the table view. That closure only calls a method that ultimately creates an array of `SiteDesignSection`, which is what we ultimately need in `SiteDesignContentCollectionViewController`. Given all that, wouldn't it be simpler if we fetch those sections here, and then send a message to `SiteDesignContentCollectionViewController` when the fetch is completed? We could do this using `WordPressFlux` and maybe we could also spare the use of the boolean `isLoading`? Thoughts?

Reply:
@Gio2018 I've pushed up 07dc22c68036f940d1a4f678277b8e3a25788e8a that simplifies things some, but still uses the builder closure.